packages/blocks/date-helper/src/lib/common/index.ts

```diff
@@ -448,38 +451,31 @@ export function addSubtractTime(date: Date, expression: string) {
       units.push(unit);
     }
   }
+
+  const validUnits = new Set([
+    'year',
+    'month',
+    'day',
+    'hour',
+    'minute',
+    'second',
+  ]);
+
   for (let i = 0; i < numbers.length; i++) {
-    const val = units[i].toLowerCase() as timeParts;
-    switch (val) {
-      case timeParts.year:
-        date.setFullYear(date.getFullYear() + numbers[i]);
-        break;
-      case timeParts.month:
-        date.setMonth(date.getMonth() + numbers[i]);
-        break;
-      case timeParts.day:
-        date.setDate(date.getDate() + numbers[i]);
-        break;
-      case timeParts.hour:
-        date.setHours(date.getHours() + numbers[i]);
-        break;
-      case timeParts.minute:
-        date.setMinutes(date.getMinutes() + numbers[i]);
-        break;
-      case timeParts.second:
-        date.setSeconds(date.getSeconds() + numbers[i]);
-        break;
-      case timeParts.dayOfWeek:
-      case timeParts.monthName:
-      case timeParts.unix_time:
-        break;
-      default: {
-        const nvr: never = val;
-        console.error(nvr, 'unhandled case was reached');
+    const value = numbers[i];
+    const unit = units[i].toLowerCase();
+
+    if (validUnits.has(unit)) {
+      const manipulateUnit = unit as dayjs.ManipulateType;
+      if (value > 0) {
+        dayjsDate = dayjsDate.add(value, manipulateUnit);
```

Comment:
@ravikiranvm @bigfluffycookie just always do
`dayjsDate = dayjsDate.add(value, manipulateUnit);`

if the value is negative it will act as subtract. 
Also, please add tests.